### PR TITLE
RCAL-878: TVAC Node Saving

### LINKED ISSUE #361: Romancal Cannot Save Intermediate Outputs from TVAC Files
Trying to run `romancal.dq_init.DQInitStep` on a TVAC file and then saving the result produces an RepresenterError exception:

```
---------------------------------------------------------------------------
RepresenterError                          Traceback (most recent call last)
Cell In[1], line 20
     17 tvac_path = '/user/sbetti/roman_tools/'
     18 tvac_file2 = tvac_path + 'irrc_test_WFI11_uncal_'
---> 20 result = DQInitStep.call(input_data, save_results=True, output_file="RGC_DQstep_dqinit.asdf")

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/stpipe/step.py:653, in Step.call(cls, *args, **kwargs)
    650 name = config.get("name", None)
    651 instance = cls.from_config_section(config, name=name, config_file=config_file)
--> 653 return instance.run(*args)

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/stpipe/step.py:520, in Step.run(self, *args)
    518     idx = None
    519 if isinstance(result, AbstractDataModel):
--> 520     self.save_model(result, idx=idx, format=self.name_format)
    521 elif hasattr(result, "save"):
    522     try:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/stpipe/step.py:992, in Step.save_model(self, model, suffix, idx, output_file, force, format, **components)
    990         output_file = model.meta.filename
    991         idx = None
--> 992     output_path = model.save(
    993         self.make_output_path(
    994             basepath=output_file,
    995             suffix=suffix,
    996             idx=idx,
    997             name_format=format,
    998             **components,
    999         )
   1000     )
   1001     self.log.info(f"Saved model in {output_path}")
   1003 return output_path

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/roman_datamodels/datamodels/_core.py:216, in DataModel.save(self, path, dir_path, *args, **kwargs)
    214 # TODO: Support gzip-compressed fits
    215 if ext == ".asdf":
--> 216     self.to_asdf(output_path, *args, **kwargs)
    217 else:
    218     raise ValueError(f"unknown filetype {ext}")

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/roman_datamodels/datamodels/_core.py:235, in DataModel.to_asdf(self, init, *args, **kwargs)
    233 print(tree)
    234 asdf_file = asdf.AsdfFile(tree)
--> 235 asdf_file.write_to(init)

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/_asdf.py:1331, in AsdfFile.write_to(self, fd, all_array_storage, all_array_compression, compression_kwargs, pad_blocks, include_block_index, version)
   1329 try:
   1330     with generic_io.get_file(fd, mode="w") as fd:
-> 1331         self._serial_write(fd, pad_blocks, include_block_index)
   1332 finally:
   1333     if version is not None:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/_asdf.py:1098, in AsdfFile._serial_write(self, fd, pad_blocks, include_block_index)
   1095     if "history" in self._tree:
   1096         tree["history"] = copy.deepcopy(self._tree["history"])
-> 1098     self._write_tree(tree, fd, pad_blocks)
   1099     self._blocks.write(pad_blocks, include_block_index)
   1100 finally:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/_asdf.py:1070, in AsdfFile._write_tree(self, tree, fd, pad_blocks)
   1067         else:
   1068             tagged_tree.pop("history", None)
-> 1070     yamlutil.dump_tree(
   1071         tree,
   1072         fd,
   1073         self,
   1074         tree_finalizer=_tree_finalizer,
   1075         _serialization_context=serialization_context,
   1076     )
   1078 if pad_blocks:
   1079     padding = util.calculate_padding(fd.tell(), pad_blocks, fd.block_size)

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:406, in dump_tree(tree, fd, ctx, tree_finalizer, _serialization_context)
    403             if key not in tags:
    404                 tags[key] = val
--> 406 yaml.dump_all(
    407     [tree],
    408     stream=fd,
    409     Dumper=AsdfDumper,
    410     explicit_start=True,
    411     explicit_end=True,
    412     version=_YAML_VERSION,
    413     allow_unicode=True,
    414     encoding="utf-8",
    415     tags=tags,
    416 )

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/__init__.py:241, in dump_all(documents, stream, Dumper, default_style, default_flow_style, canonical, indent, width, allow_unicode, line_break, encoding, explicit_start, explicit_end, version, tags, sort_keys)
    239     dumper.open()
    240     for data in documents:
--> 241         dumper.represent(data)
    242     dumper.close()
    243 finally:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:27, in BaseRepresenter.represent(self, data)
     26 def represent(self, data):
---> 27     node = self.represent_data(data)
     28     self.serialize(node)
     29     self.represented_objects = {}

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:39, in AsdfDumper.represent_data(self, data)
     38 def represent_data(self, data):
---> 39     node = super().represent_data(data)
     41     tag_name = getattr(data, "_tag", None)
     42     if tag_name is not None:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:48, in BaseRepresenter.represent_data(self, data)
     46 data_types = type(data).__mro__
     47 if data_types[0] in self.yaml_representers:
---> 48     node = self.yaml_representers[data_types[0]](self, data)
     49 else:
     50     for data_type in data_types:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:59, in represent_mapping(dumper, mapping)
     57 def represent_mapping(dumper, mapping):
     58     flow_style = _flow_style_map.get(mapping.flow_style, None)
---> 59     node = super(AsdfDumper, dumper).represent_mapping(None, mapping.data, flow_style)
     61     if mapping.property_order:
     62         values = node.value

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:118, in BaseRepresenter.represent_mapping(self, tag, mapping, flow_style)
    116 for item_key, item_value in mapping:
    117     node_key = self.represent_data(item_key)
--> 118     node_value = self.represent_data(item_value)
    119     if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120         best_style = False

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:39, in AsdfDumper.represent_data(self, data)
     38 def represent_data(self, data):
---> 39     node = super().represent_data(data)
     41     tag_name = getattr(data, "_tag", None)
     42     if tag_name is not None:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:48, in BaseRepresenter.represent_data(self, data)
     46 data_types = type(data).__mro__
     47 if data_types[0] in self.yaml_representers:
---> 48     node = self.yaml_representers[data_types[0]](self, data)
     49 else:
     50     for data_type in data_types:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:59, in represent_mapping(dumper, mapping)
     57 def represent_mapping(dumper, mapping):
     58     flow_style = _flow_style_map.get(mapping.flow_style, None)
---> 59     node = super(AsdfDumper, dumper).represent_mapping(None, mapping.data, flow_style)
     61     if mapping.property_order:
     62         values = node.value

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:118, in BaseRepresenter.represent_mapping(self, tag, mapping, flow_style)
    116 for item_key, item_value in mapping:
    117     node_key = self.represent_data(item_key)
--> 118     node_value = self.represent_data(item_value)
    119     if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120         best_style = False

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/asdf/yamlutil.py:39, in AsdfDumper.represent_data(self, data)
     38 def represent_data(self, data):
---> 39     node = super().represent_data(data)
     41     tag_name = getattr(data, "_tag", None)
     42     if tag_name is not None:

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:58, in BaseRepresenter.represent_data(self, data)
     56     node = self.yaml_multi_representers[None](self, data)
     57 elif None in self.yaml_representers:
---> 58     node = self.yaml_representers[None](self, data)
     59 else:
     60     node = ScalarNode(None, str(data))

File /user/sbetti/linux/miniconda3/envs/wfi_rfp/lib/python3.10/site-packages/yaml/representer.py:231, in SafeRepresenter.represent_undefined(self, data)
    230 def represent_undefined(self, data):
--> 231     raise RepresenterError("cannot represent an object", data)

RepresenterError: ('cannot represent an object', {'Activity Number': '5', 'Exposure Number': '1', 'Exposure QLPATH': '/common/gs448wfi/dumpfolder/NOM_OPS/OTP00614_Band2_TV1a_R1_MCEA/A5_exp1', 'GMT': '23-291-19:30:13.000', 'GSORC_BB_RAMPRATE': '0', 'GSORC_FAMXPOS': '-0.00225', 'GSORC_FAMYPOS': '-4.99784', 'GSORC_FAMZPOS': '-0.001583', 'GSORC_FAM_MOVING': '0', 'GSORC_PIC_HEADER_MATF': '0', 'GSORC_PIC_HEADER_TARGET': '0', 'GSORC_POSE_MODE': '5DOF_IN', 'GSORC_POSE_TARGET': 'BLACKBODY', 'GSORC_SDS_ACTIVE_ATTEN': '21.979', 'GSORC_SDS_DAQ_DELAY': '0.001', 'GSORC_SDS_DAQ_PINTERVAL': '65000', 'GSORC_SDS_DAQ_PULSE': '1', 'GSORC_SDS_DAQ_PW': '3', 'GSORC_SDS_FS_CONFIGURATION': '55', 'GSORC_SDS_LLTFIR_WAVE': '1147', 'GSORC_SDS_LLTFVIS_WAVE': '800.009', 'GSORC_SDS_SLCTIR_TABLEROW': '0', 'GSORC_SDS_SLCTVIS_TABLEROW': '19', 'GSORC_SDS_SORC_ON': '0', 'GSORC_SDS_SORC_WLEN': '1550', 'GSORC_TCM_07_HEATER1_SETPT': '220', 'W399EWA_FILTER_POS': 'HOME_DARK', 'WFIF_FPE_LASTGWEXPINDEX[1]': '1', 'WFIF_FPE_LASTSCIEXPINDEX[1]': '71', 'WFI_MCE_SRCS_BANK1SEL_DUMMY': '1', 'WFI_MCE_SRCS_BANK1SEL_LED1': '0', 'WFI_MCE_SRCS_BANK1SEL_LED2': '0', 'WFI_MCE_SRCS_BANK1SEL_LED3': '0', 'WFI_MCE_SRCS_BANK1SEL_LED4': '0', 'WFI_MCE_SRCS_BANK1SEL_LED5': '0', 'WFI_MCE_SRCS_BANK1SEL_LED6': '0', 'WFI_MCE_SRCS_BANK1_LED_I': '147', 'WFI_MCE_SRCS_BANK2SEL_DUMMY': '1', 'WFI_MCE_SRCS_BANK2SEL_LED1': '0', 'WFI_MCE_SRCS_BANK2SEL_LED2': '0', 'WFI_MCE_SRCS_BANK2SEL_LED3': '0', 'WFI_MCE_SRCS_BANK2SEL_LED4': '0', 'WFI_MCE_SRCS_BANK2SEL_LED5': '0', 'WFI_MCE_SRCS_BANK2SEL_LED6': '0', 'WFI_MCE_SRCS_BANK2_LED_I': '272', 'WFI_OTP_ACTNUM': '5', 'WFI_OTP_ADFNAME': 'OTP00614_Band2_TV1a_R1.txt', 'WFI_OTP_DROPFRAMES': '0', 'WFI_OTP_ENVIRONMENT': 'Not Set', 'WFI_OTP_EXPDURATION': '306', 'WFI_OTP_FRAMETIME': '0', 'WFI_OTP_GWMA': '1', 'WFI_OTP_GWMATABLE': 'WIM_TRACK', 'WFI_OTP_GWXSIZE_00': '16', 'WFI_OTP_GWXSIZE_01': '0', 'WFI_OTP_GWXSIZE_02': '0', 'WFI_OTP_GWXSIZE_03': '0', 'WFI_OTP_GWXSIZE_04': '0', 'WFI_OTP_GWXSIZE_05': '0', 'WFI_OTP_GWXSIZE_06': '0', 'WFI_OTP_GWXSIZE_07': '0', 'WFI_OTP_GWXSIZE_08': '0', 'WFI_OTP_GWXSIZE_09': '0', 'WFI_OTP_GWXSIZE_10': '0', 'WFI_OTP_GWXSIZE_11': '0', 'WFI_OTP_GWXSIZE_12': '0', 'WFI_OTP_GWXSIZE_13': '0', 'WFI_OTP_GWXSIZE_14': '0', 'WFI_OTP_GWXSIZE_15': '0', 'WFI_OTP_GWXSIZE_16': '0', 'WFI_OTP_GWXSIZE_17': '0', 'WFI_OTP_GWXSIZE_18': '0', 'WFI_OTP_GWXSTART_00': '16', 'WFI_OTP_GWXSTART_01': '0', 'WFI_OTP_GWXSTART_02': '0', 'WFI_OTP_GWXSTART_03': '0', 'WFI_OTP_GWXSTART_04': '0', 'WFI_OTP_GWXSTART_05': '0', 'WFI_OTP_GWXSTART_06': '0', 'WFI_OTP_GWXSTART_07': '0', 'WFI_OTP_GWXSTART_08': '0', 'WFI_OTP_GWXSTART_09': '0', 'WFI_OTP_GWXSTART_10': '0', 'WFI_OTP_GWXSTART_11': '0', 'WFI_OTP_GWXSTART_12': '0', 'WFI_OTP_GWXSTART_13': '0', 'WFI_OTP_GWXSTART_14': '0', 'WFI_OTP_GWXSTART_15': '0', 'WFI_OTP_GWXSTART_16': '0', 'WFI_OTP_GWXSTART_17': '0', 'WFI_OTP_GWXSTART_18': '0', 'WFI_OTP_GWYSIZE_00': '16', 'WFI_OTP_GWYSIZE_01': '0', 'WFI_OTP_GWYSIZE_02': '0', 'WFI_OTP_GWYSIZE_03': '0', 'WFI_OTP_GWYSIZE_04': '0', 'WFI_OTP_GWYSIZE_05': '0', 'WFI_OTP_GWYSIZE_06': '0', 'WFI_OTP_GWYSIZE_07': '0', 'WFI_OTP_GWYSIZE_08': '0', 'WFI_OTP_GWYSIZE_09': '0', 'WFI_OTP_GWYSIZE_10': '0', 'WFI_OTP_GWYSIZE_11': '0', 'WFI_OTP_GWYSIZE_12': '0', 'WFI_OTP_GWYSIZE_13': '0', 'WFI_OTP_GWYSIZE_14': '0', 'WFI_OTP_GWYSIZE_15': '0', 'WFI_OTP_GWYSIZE_16': '0', 'WFI_OTP_GWYSIZE_17': '0', 'WFI_OTP_GWYSIZE_18': '0', 'WFI_OTP_GWYSTART_00': '16', 'WFI_OTP_GWYSTART_01': '0', 'WFI_OTP_GWYSTART_02': '0', 'WFI_OTP_GWYSTART_03': '0', 'WFI_OTP_GWYSTART_04': '0', 'WFI_OTP_GWYSTART_05': '0', 'WFI_OTP_GWYSTART_06': '0', 'WFI_OTP_GWYSTART_07': '0', 'WFI_OTP_GWYSTART_08': '0', 'WFI_OTP_GWYSTART_09': '0', 'WFI_OTP_GWYSTART_10': '0', 'WFI_OTP_GWYSTART_11': '0', 'WFI_OTP_GWYSTART_12': '0', 'WFI_OTP_GWYSTART_13': '0', 'WFI_OTP_GWYSTART_14': '0', 'WFI_OTP_GWYSTART_15': '0', 'WFI_OTP_GWYSTART_16': '0', 'WFI_OTP_GWYSTART_17': '0', 'WFI_OTP_GWYSTART_18': '0', 'WFI_OTP_NREADFRAMES': '99', 'WFI_OTP_NUMFRAMES': '100', 'WFI_OTP_NUMRESULTANTS': '100', 'WFI_OTP_SCIMA': '71', 'WFI_OTP_SCIMATABLE': 'FULLMA510ALL', 'WFI_OTP_SORC_BB_RAMP': '0', 'WFI_OTP_SORC_BB_SETP': '0', 'WFI_OTP_SORC_DITHER_VEL': '0', 'WFI_OTP_SORC_DITHER_X': '0', 'WFI_OTP_SORC_DITHER_Y': '0', 'WFI_OTP_SORC_FAM_Z': '99', 'WFI_OTP_SORC_ILLUMINATE': 'Not Set', 'WFI_OTP_SORC_LLTF_AMP': '0', 'WFI_OTP_SORC_LLTF_WL': '0', 'WFI_OTP_SORC_MODE': 'Not Set', 'WFI_OTP_SORC_PERIOD': '0', 'WFI_OTP_SORC_PIC_MATF': 'Not Set', 'WFI_OTP_SORC_PIC_WFI': 'Not Set', 'WFI_OTP_SORC_SDS_ATTEN': 'Not Set', 'WFI_OTP_SORC_SDS_CFG': 'Not Set', 'WFI_OTP_SORC_SDS_DELAY': '0', 'WFI_OTP_SORC_SDS_FIANIUM': 'Not Set', 'WFI_OTP_SORC_SDS_FIANIUM_WL': '0', 'WFI_OTP_SORC_SDS_MODE': '99', 'WFI_OTP_SORC_SDS_PINTERVAL': 'Not Set', 'WFI_OTP_SORC_SDS_PW': '0', 'WFI_OTP_SORC_TARGET': 'Not Set', 'WFI_OTP_SRCS_LEDB1': 'Not Set', 'WFI_OTP_SRCS_LEDB1_FLUX': '0', 'WFI_OTP_SRCS_LEDB1_SETTLE': 'Not Set', 'WFI_OTP_SRCS_LEDB2': 'Not Set', 'WFI_OTP_SRCS_LEDB2_FLUX': '0', 'WFI_OTP_SRCS_LEDB2_SETTLE': 'Not Set', 'WFI_OTP_TARGETTYPE': 'DARK', 'WFI_OTP_TESTNAME': 'Not Set', 'WFI_OTP_TESTPHASE': 'NOM_OPS', 'WFI_OTP_WFI_ACM': 'Not Set', 'WFI_OTP_WFI_EWA': 'Not Set', 'WFI_OTP_WFI_GWID': '00614020010050010011', 'WFI_OTP_WFI_NEXP': '3', 'WFI_OTP_WFI_SCIID': '00614020010050010010110100000000', 'WFI_OTP_WFI_WAIT_AFTEREXP': '0', 'WFI_OTP_WFI_WAIT_BEFOREEXP': '0', 'WFI_SCE_1_GW_Xsize': '16', 'WFI_SCE_1_GW_Ysize': '16', 'WFI_SCE_1_SciReadFrames': '99', 'WFI_SCE_1_SciRstReadFrames': '1', 'WFI_TPT_GWXSIZE': '16', 'WFI_TPT_GWYSIZE': '16', 'filename': '/explore/nobackup/projects/romanst/wfi_sdp/sdp_output/WFI_TVAC1/NOM_OPS/OTP00614_Band2_TV1a_R1_MCEA/20231202T082047Z-A5_exp1-L1a/Telemetry/wfi_tm_20231018193004_02.h5', 'otp_csv_name': '/css/romanst/archive/SDH/WFI_TVAC1/NOM_OPS/OTP00614_Band2_TV1a_R1_MCEA/OTP00614_Band2_TV1a_R1_Quicklook_23-291.csv', 'product_date': '2023-12-02T08:21:18.573Z', 'product_version': '3.1', 'sample_cvt_name': '/css/romanst/archive/SDH/WFI_TVAC1/NOM_OPS/OTP00614_Band2_TV1a_R1_MCEA/A5_exp1/rst_sample_cvt_input_ball.cvt_rt_exp1_2023-291-19-30-04.cvt'})
```

I've made a workaround for this:

```
from romancal.dq_init import DQInitStep

dq = DQInitStep()
res = dq.call('TVAC1_NOMOPS_WFIRCS_20231018193013_WFI11_uncal.asdf')

# Delete TVAC stuff that can't be saved
res.__delitem__('input_meta')
res.__delitem__('hdf5_telemetry_attrs')
meta_items = 'groundtest gw_meta hdf5_meta hdf5_telemetry statistics'.split()
for mi in meta_items:
     res['meta'].__delitem__(mi)

res.save('TVAC1_NOMOPS_WFIRCS_20231018193013_WFI11_dq.asdf')
```

The input file for this is located at `/grp/roman/desjardins/TVAC1_NOMOPS_WFIRCS_20231018193013_WFI11_uncal.asdf` I'm attaching my environment here.
[env.txt](https://github.com/user-attachments/files/16288414/env.txt)

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RCAL-878](https://jira.stsci.edu/browse/RCAL-878)

Closes #361 

This PR addresses extra TVAC specific data being saved in RomanCAL.

**Checklist**
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [ ] updated relevant documentation
- [x] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://github.com/spacetelescope/RegressionTests/actions/runs/10202394832

### CODE REVIEW

**Comment by PaulHuwe** [comment]
The 83 failures in regression tests are due to two errors:
- meta.background is dealt like a stnode (as opposed to a dict) despite lacking a schema. This is something I will be working next, and thus not a concern.
- ['meta']['photometry'] may have changed type from stnode.Photometry to dict for some files.. I will investigate further.

File: CHANGES.rst [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 - Enable asdf "lazy_tree" mode for all roman datamodels files [#358]
 
+- Fix to preserve Tvac & FPS tagged scalar node types. [#368]
+
```
Comment (braingram) [inline CHANGES.rst:11]:
```suggestion
```

**Review by braingram — APPROVED** [review]
LGTM. Just a minor changelog change (and precommit is complaining).

File: tests/test_models.py [hunk 2]
```diff
@@ -970,17 +970,56 @@ def test_datamodel_save_filename(tmp_path):
...
+    model.meta["filename"] = "Roman_keys_test.asdf"
+    model2.meta.filename = "Roman_dot_test.asdf"
+
+    assert model.validate() is None
+    assert model2.validate() is None
+
+    # Test assignment types
+    assert type(model["meta"]["filename"]) == type(model2.meta["filename"])
+    assert type(model["meta"]["filename"]) == type(model2.meta.filename)
+    assert type(model.meta.filename) == type(model2.meta["filename"])
```
Comment (braingram) [inline tests/test_models.py:1025]:
```suggestion
    assert type(model["meta"]["filename"]) == type(model2.meta["filename"])  # noqa: E721
    assert type(model["meta"]["filename"]) == type(model2.meta.filename)  # noqa: E721
    assert type(model.meta.filename) == type(model2.meta["filename"])  # noqa: E721
```

File: tests/test_models.py [hunk 3]
```diff
@@ -970,17 +970,56 @@ def test_datamodel_save_filename(tmp_path):
...
+    model = utils.mk_datamodel(
+        model_class, meta={"calibration_software_version": "1.2.3", "exposure": {"read_pattern": [[1], [2], [3]]}}
+    )
+
+    assert model["meta"]["filename"] == model.meta["filename"]
+    assert model["meta"]["filename"] == model.meta.filename
+    assert model.meta.filename == model.meta["filename"]
+    assert type(model["meta"]["filename"]) == type(model.meta["filename"])
+    assert type(model["meta"]["filename"]) == type(model.meta.filename)
+    assert type(model.meta.filename) == type(model.meta["filename"])
```
Comment (braingram) [inline tests/test_models.py:1011]:
```suggestion
    assert type(model["meta"]["filename"]) == type(model.meta["filename"])  # noqa: E721
    assert type(model["meta"]["filename"]) == type(model.meta.filename)  # noqa: E721
    assert type(model.meta.filename) == type(model.meta["filename"])  # noqa: E721
```

**Pull request merged**